shared-bindings/tilepalettemapper/TilePaletteMapper.c

```diff
@@ -21,33 +22,38 @@
 //|     bitmap with a wider array of colors."""
 //|
 //|     def __init__(
-//|         self, palette: displayio.Palette, input_color_count: int, width: int, height: int
+//|         self, palette: displayio.Palette, input_color_count: int, tilegrid: displayio.TileGrid
```

Comment:
I wonder if it makes more sense to reverse this and make TPM not take width, height or tilegrid. That way you could create the TPM and pass it to a TG. The TG can then init it. That makes it fit better within the pixel_shader model I think.